integration/hibernate-4/src/main/java/com/blazebit/persistence/impl/hibernate/Hibernate4Integrator.java

```diff
@@ -36,7 +37,8 @@ public void integrate(Configuration configuration, SessionFactoryImplementor ses
 				// TODO: check that no collections are mapped
 			}
 		}
-		CommonHibernateUtils.setCollectionPersisterClass(configuration.getClassMappings(), CustomOneToManyPersister.class);
+
+		serviceRegistry.locateServiceBinding(PersisterClassResolver.class).setService(new CustomPersisterClassResolver());
```

Comment:
Perhaps this is not the best way to provide Hibernate services, as I am not really familiar with the SPI. Any input?